src/VideoCommand.cc

```diff
@@ -0,0 +1,352 @@
+    }
+
+    return VCL::Video::Codec::NOCODEC;
+}
+
+//========= AddVideo definitions =========
+
+AddVideo::AddVideo() : VideoCommand("AddVideo")
+{
+    _storage_video = VDMSConfig::instance()->get_path_videos();
+}
+
+int AddVideo::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    int node_ref = get_value<int>(cmd, "_ref",
+                                  query.get_available_reference());
+
+    VCL::Video video((void*)blob.data(), blob.size());
+
+    if (cmd.isMember("operations")) {
+        enqueue_operations(video, cmd["operations"]);
+    }
+
+    // The container and codec are checked by the schema.
+    // We default to mp4 and h264, if not specified
+    std::string container  = get_value<std::string>(cmd, "container", "mp4");
+    std::string file_name  = VCL::create_unique(_storage_video, container);
+
+    // Modifiyng the existing properties that the user gives
+    // is a good option to make the AddNode more simple.
+    // This is not ideal since we are manupulating with user's
+    // input, but for now it is an acceptable solution.
+    Json::Value props = get_value<Json::Value>(cmd, "properties");
+    props[VDMS_VID_PATH_PROP] = file_name;
+
+    // Add Video node
+    query.AddNode(node_ref, VDMS_VID_TAG, props, Json::Value());
+
+    std::string codec = get_value<std::string>(cmd, "codec", "h264");
+    VCL::Video::Codec vcl_codec = string_to_codec(codec);
+
+    video.store(file_name, vcl_codec);
+
+    // In case we need to cleanup the query
+    error["video_added"] = file_name;
+
+    if (cmd.isMember("link")) {
+        add_link(query, cmd["link"], node_ref, VDMS_VID_EDGE);
+    }
+
+    return 0;
+}
+
+//========= UpdateImage definitions =========
+
+UpdateVideo::UpdateVideo() : VideoCommand("UpdateVideo")
+{
+}
+
+int UpdateVideo::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    int node_ref = get_value<int>(cmd, "_ref", -1);
+
+    Json::Value constraints = get_value<Json::Value>(cmd, "constraints");
+
+    Json::Value props = get_value<Json::Value>(cmd, "properties");
+
+    Json::Value remove_props = get_value<Json::Value>(cmd, "remove_props");
+
+    // Update Image node
+    query.UpdateNode(node_ref, VDMS_VID_TAG, props,
+                        remove_props,
+                        constraints,
+                        get_value<bool>(cmd, "unique", false));
+
+    return 0;
+}
+
+Json::Value UpdateVideo::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    assert(responses.size() == 1);
+
+    Json::Value ret;
+
+    // TODO In order to support "codec" or "operations", we could
```

Comment:
Is this consistent with what images let you do for update?

Reply:
yeap, would be nice to add for later.